allennlp/training/__init__.py

```diff
@@ -9,3 +9,10 @@
     TrainerCallback,
     TrackEpochCallback,
 )
+from allennlp.training.deepspeed import DeepspeedTrainer
+
+# import warnings
+# try:
+#     from allennlp.training.deepspeed import DeepspeedTrainer
+# except ImportError:
+#     warnings.warn('Deepspeed plugin not installed. Ignoring.')
```

Comment:
Is this leftover debug code?

Reply:
Depends on how we want to include this. Based on my experience, I wouldn't recommend making deepspeed a required dependency. If we're doing the `pip install allennlp[deepspeed]` thing, this could be replaced/updated (not sure offhand how that gets handled but I can look for some examples).

Reply:
If you don't mind doing the work making it optional, then let's make it optional.

Reply:
> Is there a way to detect whether we are in a deepspeed context? If so, I'd be OK with some sort of `if not in_deepspeed:`. Otherwise, let's just duplicate it.

I mean, one easy way would just be to set an environment variable `DEEPSPEED=1` :stuck_out_tongue: Would that work, or would you rather more a more general solution?